### PR TITLE
deps(robots-parser): patch robots-parser to work in both node and browser env

### LINKED ISSUE #4794: Page is blocked from indexing (libUrl.parse is not a function)
Hi,

Get this fail on every website. 
By the way clicking "Learn more" becomes 404.
indexing

### PR DESCRIPTION
**Pull request by kdzwinel**
I've tried creating a browserify transform that replaces `request('url')` calls with `request('path/to/url-shim.js')`. It worked, but uncovered another issue - robots-parser is using `url.path` that doesn't exists on the web and is deprecated in node. So I decied to go with https://github.com/ds300/patch-package as suggested by @patrickhulce and @wardpeet.

Fixes #4794, but nothing will show up in the details table due to #4818

### CODE REVIEW

File: patches/robots-parser+1.0.2.patch [hunk 1]
```diff
@@ -0,0 +1,36 @@
+patch-package
+--- a/node_modules/robots-parser/Robots.js
```
Comment (kdzwinel) [inline patches/robots-parser+1.0.2.patch]: a diff of a diff 😵

**Comment by paulirish** [comment]
Presumably this also needs some post install script to apply the patch?

**Comment by kdzwinel** [comment]
@paulirish I also thought so, but …

> Patches created by patch-package are automatically and gracefully applied when you use npm(>=5) or yarn.

🕺

**Comment by wardpeet** [comment]
Yeah patch-package is awesome!

**Comment by kdzwinel** [comment]
Yeah, I agree it's not perfect. I thought about opening a PR with this patch against the original repo. We can also pull that library into LH or fork it.

**Comment by patrickhulce** [comment]
>  I thought about opening a PR with this patch against the original repo

PR to have robots-parser use real `URL` class and switch to have us swap `require('url')` with our shim sounds like most reasonable long term solution to me

**Comment by paulirish** [comment]
+1 to giving @samclarke a PR. 
Hi Sam, big fan. :)

**Comment by kdzwinel** [comment]
Perfect, I'll open a PR👍 should I create an issue to track this?

**Comment by brendankenny** [comment]
> PR to have robots-parser use real URL class and switch to have us swap require('url') with our shim sounds like most reasonable long term solution to me

SGreatTM

**Comment by brendankenny** [comment]
> should I create an issue to track this?

that SG too

**Comment by kdzwinel** [comment]
Issue created (#4825) 👌

**Pull request merged**